src/integer_mod_q/mat_zq/solve.rs

```diff
@@ -1,28 +1,36 @@
-// Copyright © 2023 Marvin Beckmann
+// Copyright © 2023 Marcel Luca Schmidt, Marvin Beckmann
 //
 // This file is part of qFALL-math.
 //
 // qFALL-math is free software: you can redistribute it and/or modify it under
 // the terms of the Mozilla Public License Version 2.0 as published by the
 // Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
 
-//! This module contains implementations to solve linear equations over [`MatZq`]
-//! with arbitrary moduli.
+//! This module contains implementations to solve systems of linear equations
+//! over [`MatZq`] with arbitrary moduli.
 
 use super::MatZq;
 use crate::{
-    integer::Z,
+    integer::{MatZ, Z},
     integer_mod_q::Zq,
-    traits::{Concatenate, GetEntry, GetNumColumns, GetNumRows, Pow, SetEntry},
+    traits::{Concatenate, Gcd, GetEntry, GetNumColumns, GetNumRows, Pow, SetEntry, Xgcd},
+    utils::Factorization,
 };
 
 impl MatZq {
-    /// Computes a solution for a system of linear equations under a certain modulus
-    /// It solves `Ax = y` for `x` as a [MatZq].
-    /// If no solution is found, `None` is returned
+    /// Computes a solution for a system of linear equations under a certain modulus.
+    /// It solves `Ax = y` for `x` with `A` being a [`MatZq`] value.
+    /// If no solution is found, `None` is returned.
+    /// The function uses Gaussian elimination together with Factor refinement
+    /// to split the modulus and the Chinese remainder theorem and Hensel lifting
+    /// to combine solutions under the split modulus.
+    /// For Hensel lifting we use the method from [\[1\]](<../index.html#:~:text=[1]>).
```

Comment:
Please also fix for the others, such that the links actually work
```suggestion
    /// For Hensel lifting we use the method from [\[1\]](<index.html#:~:text=[1]>).
```